datastore/shared/services/environment_service.py

```diff
@@ -39,11 +39,8 @@ def is_dev_mode(self) -> bool:
         value = self.try_get(DATASTORE_DEV_MODE_ENVIRONMENT_VAR)
         return value is not None and value.lower() in ("1", "on", "true")
 
-    def get_from_file(self, name: str) -> str:
-        path = os.environ.get(name + "_FILE", None)
-        if self.is_dev_mode():
-            self.cache[name] = DEV_SECRET
-        elif path is not None:
-            with open(path) as file_:
-                self.cache[name] = file_.read()
-        return self.get(name)
+    def get_from_file(self, name: str, use_default_secret: bool = True) -> str:
+        if self.is_dev_mode() and use_default_secret:
+            return DEV_SECRET
+        with open(self.get(name)) as file_:
```

Comment:
This change is ok for me but I do not understand it. The `get()` method uses an internal cache for all env vars. The `get_from_file()` may also cache the password read from file so it does not have to read it from the file again.

If this does not matter because this is only done one server start maybe remove this caching at all.

But as I wrote, this way is also ok for me.

Reply:
There was a reason for this caching, but I don't quite remember it. The problem with writing the secret content into the cache is that you could potentially have a name clash with a "normal" env var. So you would need an additional cache just for this case and I found that to be unnecessary and not providing enough advantage.